### PR TITLE
Linear phase signal generation

### PR DESCRIPTION
**Pull request by f-brinkmann**
### Which issue(s) are closed by this pull request?

Nothing

### Changes proposed in this pull request:

- add function and testing for linear phase signal generation

### CODE REVIEW

File: pyfar/dsp/dsp.py [hunk 1]
```diff
@@ -143,6 +143,51 @@ def wrap_to_2pi(x):
     return x
 
 
+def linear_phase(signal, group_delay=None):
```
Comment (mberz) [inline pyfar/dsp/dsp.py:146]: I am not a fan of having a group delay which is None by default, this is potentially confusing implicit behavior

File: pyfar/dsp/dsp.py [hunk 2]
```diff
@@ -143,6 +143,51 @@ def wrap_to_2pi(x):
...
+    """
+    Calculate linear phase signal.
+
+    The linear phase signal is computed as
```
Comment (mberz) [inline pyfar/dsp/dsp.py:150]: This description can be a bit confusing, as one might expect that this function calculates sets the phase to "only" the already existing linear phase part of the singal, i.e. a already pre-existing constant group delay.

File: pyfar/dsp/dsp.py (hunk 1 above)
Comment (f-brinkmann) [reply pyfar/dsp/dsp.py:146]: I'm not sure if this works, if yes it can be `group_delay=signal.n_samples/2`

File: pyfar/dsp/dsp.py (hunk 2 above)
Comment (mberz) [reply pyfar/dsp/dsp.py:150]: Set the phase to a linear phase with group delay ...

**Review by mberz — CHANGES_REQUESTED** [review]
see comments

File: pyfar/dsp/dsp.py [hunk 3]
```diff
@@ -143,6 +143,50 @@ def wrap_to_2pi(x):
     return x
 
 
+def linear_phase(signal, group_delay):
```
Comment (sikersten) [inline pyfar/dsp/dsp.py:146]: What about a unit parameter, so `group_delay` could be defined in seconds or milliseconds?

File: tests/test_dsp.py [hunk 4]
```diff
@@ -103,6 +103,24 @@ def test_group_delay_custom_frequencies(impulse_group_delay):
...
+    # test output
+    assert isinstance(y, pf.Signal)
+    npt.assert_allclose(dsp.group_delay(y), N / 2 * np.ones(y.n_bins))
+    # test if input did not change
+    npt.assert_allclose(x.time, pf.signals.impulse(N).time)
+
+    # test assertion
+    with pytest.raises(TypeError, match="signal must be a pyfar Signal"):
+        dsp.linear_phase(1, 0)
+
```
Comment (sikersten) [inline tests/test_dsp.py:122]: Missing test for multichannel signal

**Review by sikersten — CHANGES_REQUESTED** [review]

File: pyfar/dsp/dsp.py (hunk 3 above)
Comment (f-brinkmann) [reply pyfar/dsp/dsp.py:146]: Good idea, done as suggested

File: tests/test_dsp.py (hunk 4 above)
Comment (f-brinkmann) [reply tests/test_dsp.py:135]: I added testing and the possibility to pass array likes for `group_delay`

**Review by sikersten — APPROVED** [review]

**Review by mberz — APPROVED** [review]

**Pull request merged**